### PR TITLE
 source ocs-operator CSV in OCP CI 

### PR DESCRIPTION
**Pull request by jarrpa**
Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

File: hack/source-manifests.sh [hunk 1]
```diff
@@ -32,53 +32,66 @@ mkdir -p $OUTDIR_CRDS
...
+		sed -i '' "/replaces:/d" $OCS_CSV
+	else
+		sed -i "s/$TMP_CSV_VERSION/{{.OcsOperatorCsvVersion}}/g" $OCS_CSV
+		sed -i "s/REPLACE_IMAGE/{{.OcsOperatorImage}}/g" $OCS_CSV
+		sed -i "/replaces:/d" $OCS_CSV
+	fi
+	cp deploy/crds/* $OUTDIR_CRDS/
+}
+
+if [ ! -n "$OPENSHIFT_BUILD_NAMESPACE" ]; then
```
Comment (obnoxxx) [inline hack/source-manifests.sh:89]: why `if [ ! -n "..." ]` instead of `if [ -z "..." ]` ?

Comment (obnoxxx) [reply hack/source-manifests.sh:89]: heh, shellcheck confirms this: `^-- SC2236: Use -z instead of ! -n.`

**Review by obnoxxx — CHANGES_REQUESTED** [review]

File: hack/source-manifests.sh (hunk 1 above)
Comment (jarrpa) [reply hack/source-manifests.sh:89]: D'oh! Fair.

**Review by obnoxxx — CHANGES_REQUESTED** [review]
First of all, you know I like to separate refactorings (like putting chunks of code into funcitons) and making semantic changes (like *when* to run them) into separate commits. Apart from that, the code looks clean.

But it does not seem to fail in the intended way:

```
 time="2020-12-16T16:44:09Z" level=info msg="Generating CSV manifest version 9999.9999.9999"
time="2020-12-16T16:44:24Z" level=info msg="Fill in the following required fields in file ocs-operator.clusterserviceversion.yaml:\n\tspec.description\n\tspec.keywords\n\tspec.provider\n"
time="2020-12-16T16:44:24Z" level=info msg="CSV manifest generated successfully"
Manifests sourced into deploy/csv-templates directory
reading in csv at deploy/csv-templates/ocs-operator.csv.yaml.in
reading in csv at deploy/csv-templates/rook-csv.yaml.in
panic: open deploy/csv-templates/rook-csv.yaml.in: no such file or directory
goroutine 1 [running]:
text/template.Must(...)
	/usr/lib/golang/src/text/template/helper.go:23
main.unmarshalCSV(0x7ffc901b1c87, 0x25, 0xc000100c00)
	/go/src/github.com/openshift/ocs-operator/tools/csv-merger/csv-merger.go:119 +0x3e5
main.generateUnifiedCSV(0xc0001714a0)
	/go/src/github.com/openshift/ocs-operator/tools/csv-merger/csv-merger.go:427 +0x105
main.main()
	/go/src/github.com/openshift/ocs-operator/tools/csv-merger/csv-merger.go:1053 +0x25d
make: *** [Makefile:85: gen-latest-csv] Error 2 
```

It fails not finding `rook-csv.yaml.in` instead of the artificially injected ocs-op csv change.

**Review by obnoxxx — APPROVED** [review]
Great update, should we test once more with a injected error on top (in a separate PR) whether `verify-latest-csv` now fails as desired?

**Comment by jarrpa** [comment]
> Great update, should we test once more with a injected error on top (in a separate PR) whether `verify-latest-csv` now fails as desired?

Certainly, error pushed.

**Comment by jarrpa** [comment]
Failure confirmed with dummy commit: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ocs-operator/962/pull-ci-openshift-ocs-operator-master-verify-latest-csv/1339297432827072512

**Review by obnoxxx — APPROVED** [review]
/lgtm

**Pull request merged**